java/java-analysis-impl/src/com/intellij/codeInspection/localCanBeFinal/LocalCanBeFinal.java

```diff
@@ -351,6 +317,197 @@ public void visitForStatement(@NotNull PsiForStatement statement) {
     return problems;
   }
 
+  /**
+  * This method is to be used in the Java 2 Kotlin conversion to determine
+  * local variables that should be defined with "val" (implying immutability)
+  *<p>
+  * @param body the block of code to search
+  * @return a set of kotlin immutable PsiVariables
+   */
+  @Nullable
+  public Set<PsiVariable> getKotlinImmutableLocalVariablesInBlock(final PsiCodeBlock body) {
```

Comment:
Is it possible to keep Kotlin-specific behavior in the Kotlin plugin? It looks out of scope of Java plugin, and an additional burden for Java team to maintain this code.

Reply:
works for me, I moved it back but left the refactor of Java so that I could reuse some code @amaembo